source/Naturals/IdOrder.lagda

```diff
@@ -0,0 +1,400 @@
+ _≼_ {{Curly-Order-ℕ-ℕ}} = _≼ℕ_
+
+private
+ example : 0 ≼ 2
+ example = refl
+
+ example2 : ¬ (1 ≼ 0)
+ example2 = positive-not-zero 1
+
+≼-is-prop-valued : (x y : ℕ) → is-prop (x ≼ y)
+≼-is-prop-valued x y p q = ℕ-is-set p q
+
+≼-is-discrete : (x y : ℕ) → is-discrete (x ≼ y)
+≼-is-discrete x y = props-are-discrete (≼-is-prop-valued x y)
+
+≼-is-set : (x y : ℕ) → is-set (x ≼ y)
+≼-is-set x y p q = discrete-types-are-sets (≼-is-discrete x y) p q
+
+≼-dichotomous : (x y : ℕ) → (x ≼ y) ∔ (y ≼ x)
+≼-dichotomous x y = +functor (_⁻¹) (_⁻¹) (diff-cancellable x y)
+
+zero-least : (x : ℕ) → zero ≼ x
+zero-least x = refl
+
+unique-least : (x : ℕ) → x ≼ zero → x ＝ zero
+unique-least zero p = refl
+
+≼-intro : (x y m n : ℕ)
+        → (x ＝ succ m)
+        → (y ＝ succ (succ (m + n)))
+        → x ≼ y
+≼-intro x y zero zero p q = transport₂ (_≼ℕ_) (p ⁻¹) (q ⁻¹) refl
+≼-intro x y zero (succ n) p q = transport₂ (_≼ℕ_) (p ⁻¹) (q ⁻¹)
+ (ap succ (zero-least (succ (succ (0 + n)))))
+≼-intro (succ x) (succ y) (succ m) n p q = ap succ
+ (≼-intro x y m n (succ-lc p)
+  (succ-lc (q ∙ ap (succ ^ 2) (succ-left m n))))
+
+≼-refl : reflexive _≼ℕ_
+≼-refl x = ap (_+ x) (difference-is-zero x) ∙ zero-left-neutral x
+
+≼-anti : antisymmetric _≼ℕ_
+≼-anti x zero p q = unique-least x p
+≼-anti (succ x) (succ y) p q = ap succ
+ (≼-anti x y (succ-lc p) (succ-lc q))
+
+≼-trans : transitive _≼ℕ_
+≼-trans zero y z p q = zero-least z
+≼-trans (succ x) (succ y) (succ z) p q =
+ ap succ (≼-trans x y z (succ-lc p) (succ-lc q))
+
+≼-trans₂ : (x y z w : ℕ) → x ≼ y → y ≼ z → z ≼ w → x ≼ w
+≼-trans₂ x y z w p q r = ≼-trans x y w p (≼-trans y z w q r )
+
+≼-transport₂ : {x y z w : ℕ} → x ＝ z → y ＝ w → x ≼ y → z ≼ w
+≼-transport₂ p q i = transport₂ (_≼ℕ_) p q i
+
+≼-＝ : (x y : ℕ) → x ＝ y → x ≼ y
+≼-＝ x y p = transport (x ≼ℕ_) p (≼-refl x)
+
+_≼⟨_⟩_ : (x : ℕ) {y z : ℕ} → x ≼ y → y ≼ z → x ≼ z
+(_≼⟨_⟩_) x {y} {z} p q = ≼-trans x y z p q
+
+_◻ : (x : ℕ) → x ≼ x
+_◻ = ≼-refl
+
+infixr 0 _≼⟨_⟩_
+infix  1 _◻
+
+private instance
+ Reasoning-Chain-≼ℕ : Reasoning-Chain ℕ ℕ ℕ (_≼ℕ_) (_≼ℕ_) (_≼ℕ_)
+ _⸴_⊢_ {{Reasoning-Chain-≼ℕ}} = _≼⟨_⟩_
+
+ Reflexive-Order-≼ℕ : Reflexive-Order ℕ (_≼ℕ_)
+ _▨ {{Reflexive-Order-≼ℕ}} = ≼-refl
+
+\end{code}
+
+Order lemmas for the successor constructor
+
+\begin{code}
+
+≼-succ : (x : ℕ) → x ≼ succ x
+≼-succ zero = refl
+≼-succ (succ x) = ap succ (≼-succ x)
+
+≼-succ^ : (n x : ℕ) → x ≼ (succ ^ n) x
+≼-succ^ zero x = ≼-refl x
+≼-succ^ (succ n) x = ≼-trans x (succ x) (succ (rec x succ n))
+ (≼-succ x) (ap succ (≼-succ^ n x))
+
+≼-close-left : (x y : ℕ) → succ x ≼ y → x ≼ y
+≼-close-left x y i = ≼-trans x (succ x) y (≼-succ x) i
+
+≼-close-right : (x y : ℕ) → x ≼ y → x ≼ succ y
+≼-close-right x y i = ≼-trans x y (succ y) i (≼-succ y)
+
+succ-monotone : (x y : ℕ) → x ≼ y → succ x ≼ succ y
+succ-monotone x y = ap succ
+
+succ-order-injective : (x y : ℕ) → succ x ≼ succ y → x ≼ y
+succ-order-injective x y = succ-lc
+
+¬≼-succ-lc : (x y : ℕ) → ¬ (succ x ≼ succ y) → ¬ (x ≼ y)
```

Comment:
I would omit the definition in line 186. It is just the contrapositive of the previous one, and we have a definition of contrapositivity in `MLTT.Negation`.